components/TextFields/examples/TextFieldObjectiveCExample.m

```diff
@@ -0,0 +1,353 @@
+  textFieldMessage.textView.delegate = self;
+
+  self.messageController = [[MDCTextInputControllerTextArea alloc] initWithTextInput:textFieldMessage];
+
+  NSDictionary *views = @{
+                         @"name": textFieldName,
+                         @"address": textFieldAddress,
+                         @"city": textFieldCity,
+                         @"state": textFieldState,
+                         @"zip": textFieldZip,
+                         @"stateZip": stateZip,
+                         @"phone": textFieldPhone,
+                         @"message": textFieldMessage
+                         };
+  NSMutableArray <NSLayoutConstraint *> *constraints = [NSMutableArray arrayWithArray:
+       [NSLayoutConstraint constraintsWithVisualFormat:@"V:|-20-[name]-[address]-[city]-[stateZip]-[phone]-[message]-20-|"
+
+                                               options:NSLayoutFormatAlignAllLeading |
+                                                          NSLayoutFormatAlignAllTrailing
+                                               metrics:nil
+                                                 views:views]];
+  [constraints addObject:
+     [NSLayoutConstraint constraintWithItem:textFieldName
+                                  attribute:NSLayoutAttributeLeading
+                                  relatedBy:NSLayoutRelationEqual
+                                     toItem:self.view
+                                  attribute:NSLayoutAttributeLeadingMargin
+                                 multiplier:1
+                                   constant:0]];
+  [constraints addObject:
+      [NSLayoutConstraint constraintWithItem:textFieldName
+                                   attribute:NSLayoutAttributeTrailing
+                                   relatedBy:NSLayoutRelationEqual
+                                      toItem:self.view
+                                   attribute:NSLayoutAttributeTrailingMargin
+                                  multiplier:1
+                                    constant:0]];
+  [constraints addObjectsFromArray:
+     [NSLayoutConstraint constraintsWithVisualFormat:@"H:|[state(80)]-[zip]|"
+                                             options:0
+                                             metrics:nil
+                                               views:views]];
+  [constraints addObjectsFromArray:
+   [NSLayoutConstraint constraintsWithVisualFormat:@"V:|[state]|"
+                                             options:0
+                                             metrics:nil
+                                               views:views]];
+  [constraints addObjectsFromArray:
+   [NSLayoutConstraint constraintsWithVisualFormat:@"V:|[zip]|"
+                                             options:0
+                                             metrics:nil
+                                               views:views]];
+  [NSLayoutConstraint activateConstraints:constraints];
+}
+
+- (void)setupScrollView {
```

Comment:
How does this look on iPhone X? Do you need to change [`-contentInsetAdjustmentBehavior`](https://developer.apple.com/documentation/uikit/uiscrollview/2902261-contentinsetadjustmentbehavior) to account for the cut-out when in lanscape?

Reply:
Yeah, it looks fine.